flytekit/core/base_task.py

```diff
@@ -290,9 +291,8 @@ def local_execute(
                 native_types=self.get_input_types(),  # type: ignore
             )
         except TypeTransformerFailedError as exc:
-            msg = f"Failed to convert inputs of task '{self.name}':\n  {exc}"
-            logger.error(msg)
-            raise TypeError(msg) from None
+            exc.args = (f"Failed to convert inputs of task '{self.name}':\n  {exc.args[0]}",)
```

Comment:
This actually starts breaking behavior for things like `AssertionError` where someone might just do `assert val > 0` without specifying a message like `assert val > 0, "val must be greater than 0`

Reply:
Does it mean flytekit raises an exception when using `AssertionError` in the user code because there is no message in the `args`?

Reply:
Yea, we get `IndexError: tuple index out of range` when a test case throws an empty `AssertionError`.

Reply:
To be more specific, I mentioned test case only because that's how I caught this bug when migrating from `1.12.x` to `1.13.x`. The task is what is actually doing the `assert val > 0` check.

Reply:
@blaketastic2 do you want to open a PR to fix it? otherwise, I'll open one later this week.

Reply:
Thank you in advance!